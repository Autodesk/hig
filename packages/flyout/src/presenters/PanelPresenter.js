import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import { ThemeContext } from "@hig/theme-context";
import { createCustomClassNames } from "@hig/utils";

import stylesheet from "./stylesheet";

export default function PanelPresenter({
  children,
  onScroll,
  stylesheet: customStylesheet,
  ...otherProps
}) {
  const { className } = otherProps;
  const panelClassName = createCustomClassNames(className, "panel");

  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet(
          {
            transitionStatus: null,
            anchorPoint: null,
            stylesheet: customStylesheet
          },
          resolvedRoles
        );

        return (
          <div
            className={cx(css(styles.panel), panelClassName)}
            onScroll={onScroll}
          >
            {children}
          </div>
        );
      }}
    </ThemeContext.Consumer>
  );
}

PanelPresenter.propTypes = {
  children: PropTypes.node,
  onScroll: PropTypes.func,
  stylesheet: PropTypes.func
};
