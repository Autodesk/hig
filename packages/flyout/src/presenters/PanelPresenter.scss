.hig__flyout-v1__panel {
  @import "../tokens.scss";

  min-width: 200px;
  max-width: 400px;
  max-height: 360px;
  padding: $panel-padding;
  overflow-y: auto;
  -ms-overflow-style: -ms-autohiding-scrollbar;
}
