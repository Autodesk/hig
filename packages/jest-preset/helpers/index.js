/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
