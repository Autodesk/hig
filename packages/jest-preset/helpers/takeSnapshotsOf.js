const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
