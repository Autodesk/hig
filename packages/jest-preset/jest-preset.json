{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
