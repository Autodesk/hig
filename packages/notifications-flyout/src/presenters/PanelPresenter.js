import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import { Panel } from "@hig/flyout";
import ProgressRing from "@hig/progress-ring";
import ThemeContext from "@hig/theme-context";
import Typography from "@hig/typography";
import {
  UNMOUNTED,
  EXITED,
  ENTERING,
  ENTERED,
  EXITING
} from "react-transition-group/Transition";

import stylesheet from "./stylesheet";

export default function PanelPresenter({
  children,
  heading,
  innerRef,
  listMaxHeight,
  loadingTransitionState,
  onScroll,
  refListWrapper,
  stylesheet: customStylesheet
}) {
  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet(
          {
            transitionState: null,
            loadingTransitionState,
            customStylesheet
          },
          resolvedRoles
        );
        return (
          <Panel innerRef={innerRef}>
            <Typography elementType="header" style={styles.panelTitle}>
              {heading}
            </Typography>
            <section
              className={css(styles.panelContainer)}
              ref={refListWrapper}
              style={{ maxHeight: listMaxHeight }}
            >
              <div role="list" onScroll={onScroll}>
                {children}
              </div>
            </section>
            <footer className={css(styles.panelLoading)}>
              <ProgressRing size="s" />
            </footer>
          </Panel>
        );
      }}
    </ThemeContext.Consumer>
  );
}

PanelPresenter.defaultProps = {
  heading: "Notifications"
};

PanelPresenter.propTypes = {
  children: PropTypes.node,
  heading: PropTypes.node,
  innerRef: PropTypes.func.isRequired,
  listMaxHeight: PropTypes.string,
  loadingTransitionState: PropTypes.oneOf([
    UNMOUNTED,
    EXITED,
    ENTERING,
    ENTERED,
    EXITING
  ]),
  onScroll: PropTypes.func,
  refListWrapper: PropTypes.func
};
