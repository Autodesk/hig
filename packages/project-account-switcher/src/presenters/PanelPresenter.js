import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";

import { Panel } from "@weave-design/flyout";
import { createCustomClassNames } from "@weave-design/utils";

import stylesheet from "./stylesheet";

export default function PanelPresenter({
  innerRef,
  children,
  stylesheet: customStylesheet,
  ...otherProps
}) {
  const styles = stylesheet({ stylesheet: customStylesheet });
  const { className } = otherProps;
  const panelClassName = createCustomClassNames(className, "panel");

  return (
    <Panel innerRef={innerRef}>
      <div className={cx([panelClassName, css(styles.panel)])}>{children}</div>
    </Panel>
  );
}

PanelPresenter.propTypes = {
  innerRef: PropTypes.func,
  children: PropTypes.node,
  stylesheet: PropTypes.func,
};
