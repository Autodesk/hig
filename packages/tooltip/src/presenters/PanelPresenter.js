import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import { ThemeContext } from "@hig/theme-context";

import stylesheet from "./stylesheet";

export default function PanelPresenter(props) {
  const { children, innerRef, maxHeight, onScroll } = props;
  const maxHeightInPixels = maxHeight ? `${maxHeight}px` : undefined;

  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet(resolvedRoles);

        return (
          <div
            className={css(styles.panel)}
            onScroll={onScroll}
            ref={innerRef}
            style={{ maxHeight: maxHeightInPixels }}
          >
            <div className={css(styles.panelInner)}>{children}</div>
          </div>
        );
      }}
    </ThemeContext.Consumer>
  );
}

PanelPresenter.propTypes = {
  /** A required ref that's used to position the flyout */
  innerRef: PropTypes.func,
  /** The panel content */
  children: PropTypes.node,
  /** Max height of the panel */
  maxHeight: PropTypes.number,
  /** Scroll event handler */
  onScroll: PropTypes.func
};
