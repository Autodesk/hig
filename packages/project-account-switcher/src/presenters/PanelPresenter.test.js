import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import PanelPresenter from "./PanelPresenter";

describe("project-account-switcher/presenters/PanelPresenter", () => {
  function handleRef() {}

  takeSnapshotsOf(PanelPresenter, [
    {
      desc: "renders with minimal props",
      props: {
        innerRef: handleRef,
      },
    },
    {
      desc: "renders with all props",
      props: {
        children: "hello",
        innerRef: handleRef,
      },
    },
  ]);
});
