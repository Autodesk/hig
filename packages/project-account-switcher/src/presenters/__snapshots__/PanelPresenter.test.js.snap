// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-account-switcher/presenters/PanelPresenter  1`] = `
.emotion-0 {
  margin: 0;
  padding: 0 10px 10px 10px;
}

.emotion-2 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-1 {
  position: relative;
}

<div
  className="emotion-2"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    />
  </div>
</div>
`;

exports[`project-account-switcher/presenters/PanelPresenter  2`] = `
.emotion-0 {
  margin: 0;
  padding: 0 10px 10px 10px;
}

.emotion-2 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-1 {
  position: relative;
}

<div
  className="emotion-2"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    >
      hello
    </div>
  </div>
</div>
`;
